src/overlays/actors/ovl_En_GirlA/z_en_girla.h

```diff
@@ -5,11 +5,61 @@
 
 struct EnGirlA;
 
+typedef void (*EnGirlAActionFunc)(struct EnGirlA*, GlobalContext*);
+typedef void (*EnGirlADrawFunc)(struct Actor*, GlobalContext*, s32);
+typedef s32 (*EnGirlACanBuyFunc)(GlobalContext*, struct EnGirlA*);
+typedef void (*EnGirlABuyFunc)(GlobalContext*, struct EnGirlA*);
+
+typedef struct ShopItemEntry {
+    /* 0x00 */ s16 objectId;
+    /* 0x02 */ s16 getItemDrawId;
+    /* 0x04 */ EnGirlADrawFunc drawFunc;
+    /* 0x08 */ s16 params;
+    /* 0x0A */ u16 descriptionTextId;
+    /* 0x0C */ u16 choiceTextId;
+    /* 0x10 */ s32 getItemId;
+    /* 0x14 */ EnGirlACanBuyFunc canBuyFunc;
+    /* 0x18 */ EnGirlABuyFunc buyFunc;
+    /* 0x1C */ EnGirlABuyFunc buyFanfareFunc;
+} ShopItemEntry; // size = 0x20
+
 typedef struct EnGirlA {
     /* 0x000 */ Actor actor;
-    /* 0x144 */ char unk_144[0x88];
+    /* 0x144 */ char unk144[0x44];
+    /* 0x188 */ EnGirlAActionFunc actionFunc;
+    /* 0x18C */ s8 objIndex;
+    /* 0x190 */ EnGirlAActionFunc mainActionFunc;
+    /* 0x194 */ s32 isInitialized;
+    /* 0x198 */ u16 choiceTextId;
+    /* 0x19C */ s32 getItemId;
+    /* 0x1A0 */ s16 isOutOfStock;
+    /* 0x1A4 */ void (*boughtFunc)(GlobalContext*, struct EnGirlA*);
```

Comment:
Why not make a typedef for this function pointer type?

Reply:
It was more that these two had the same pointer type as the buy functions, and I couldn't think of a good name to tie all of them together. Perhaps `shopActionFunc`?

Reply:
Yes.